CorsixTH/Lua/earthquake.lua

```diff
@@ -0,0 +1,237 @@
+--[[ Copyright (c) 2024 Toby "tobylane"
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of
+this software and associated documentation files (the "Software"), to deal in
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+of the Software, and to permit persons to whom the Software is furnished to do
+so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE. --]]
+
+--! Manages all earthquakes in the world
+class "Earthquake"
+
+---@type Earthquake
+local Earthquake = _G["Earthquake"]
+
+-- All fields relating to the current or next earthquake:
+--  active (boolean) If we are currently running the warning or damage timers
+--   (after start_day of start_month is passed)
+--  next_planned (boolean) If the next earthquake has been planned/generated
+--  start_month (integer) The month the earthquake warning is triggered
+--  start_day (integer) The day of the month the earthquake warning is triggered
+--  size (integer) The amount of damage the earthquake causes (1-9)
+--  remaining_damage (integer) The amount of damage this earthquake has yet to inflict
+--  damage_timer (integer) The number of hours until the earthquake next inflicts damage if active
+--  warning_timer (integer) The number of hours left until the real damaging earthquake begins
+--  current_map_earthquake (integer) The number of earthquakes that have happened on this map so far
```

Comment:
2 space indent the comment if you want that emphasis. Line 29 is out of place though (even if you intended it as a run on of 28)